src/ray/util/process.cc

```diff
@@ -265,8 +356,14 @@ int Process::Wait() const {
         status = -1;
       }
 #else
-      (void)fd;
-      if (waitpid(pid, &status, 0) != 0) {
+      if (fd != -1) {
```

Comment:
Can you add a comment when this case is being called? (vs the waitpid case)

Reply:
Sure